Show ApplicationBannerCard all the time and add /application/last endpoint

EasySouls opened the pull request:
I created the endpoint which works similarly than the existing /application/my endpoint, only that it doesn't take the current application period into consideration.

On the frontend now the banner has 4 states:
- If the user is not logged in, there are no currently active periods and the user has never submitted an application it is not shown.
- If there is a submitted application for the active period, the card is shown, just as before
- If the user has not submitted an application, but there is an active period, the same button is shown which leads to the application form, just as before
- If there is no active period, but the user has a previous application, the card is shown in the same form like with an active period, only now it is presented to the user that in which period was the card submitted.

Closes #122

EasySouls commented:
@FearsomeRover Egyelőre a typescript errort jelez, de különben működne, csak direkt nem javítottam ki, mert észrevettem valamit. Az `/application/my` és az új endpoint is egy `Application`-t ad vissza ts szerint. 
Viszont az eddig is létező `useCurrentApplication()` hook pedig a `ApplicationEntityWithPeriod` típust rendeli hozzá. Szóval ts szerint létezik period mezője, de mivel a backendről csak egy `Application`-t kap vissza, ebből csak undefined lesz.
A kérdésem az, hogy írjam át a backendes kódot, hogy includeolja a period entity-t is, vagy az frontenden igazából nem is kell, és a `useCurrentApplication()` és az új `useLastApplication()` kaphatnak sima `Application`-t?

mozsarmate commented on apps/backend/src/application/application.service.ts:
```diff
@@ -128,6 +135,35 @@ export class ApplicationService {
+          userId: user.authSchId,
+        },
+        orderBy: {
+          createdAt: 'desc',
```
lehetne applicationPeriodEnd alapján

mozsarmate commented on apps/frontend/src/components/ui/ApplicationBannerCard.tsx:
```diff
@@ -35,8 +37,56 @@ export default function ApplicationBannerCard() {
+          <div className='flex flex-col gap-4 justify-start'>
+            <CardTitle>Eddigi jelentkezésed</CardTitle>
+            <p>
+              Jelenleg nincsen jelentkezési időszak, de itt láthatod a(z){' '}
```
szerintem ezt kivehetjük, amúgy is straight-forward

mozsarmate reviewed (approved):
Patika munka, nyomoood!

EasySouls commented:
I also refactored the types regarding the applications. Since the currentApplication wasn't using the applicationPeriod's data, I changed the swr function's return type to be simply ApplicationEntity.
And changed the lastApplication's return type to be ApplicationEntityWithPeriod, since it might be useful to get that periods information in which the last application was submitted in.
In the backend controller I corrected the return type as well, but it is quite nasty looking, and since it uses Prisma's types, it doesn't show up in the OpenAPI specs, so it might be better to change it to our own dto in the future.

EasySouls commented:
What I meant to say is that before the changes, we could get the currentApplication's period data in the frontend, but since the backend doesn't include it in the response, it would result in an error. 
And now we can typesafely get the period data using the `useLastApplication()` hook.

mozsarmate commented:
Looks good to me, nice work!
